### PR TITLE
Prepare statements support

### PR DESCRIPTION
**Pull request by tmszdmsk**
Fixes https://github.com/opentracing-contrib/java-cassandra-driver/issues/8.

One test case added. Under the hood it uses one of `prepareAsync` methods on `session`, I've fixed all `prepare*` methods though.

### CODE REVIEW

File: src/main/java/io/opentracing/contrib/cassandra/TracingSession.java [hunk 1]
```diff
@@ -13,16 +13,7 @@
...
-import com.datastax.driver.core.CloseFuture;
-import com.datastax.driver.core.Cluster;
-import com.datastax.driver.core.Host;
-import com.datastax.driver.core.PreparedStatement;
-import com.datastax.driver.core.RegularStatement;
-import com.datastax.driver.core.ResultSet;
-import com.datastax.driver.core.ResultSetFuture;
-import com.datastax.driver.core.Session;
-import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.*;
```
Comment (malafeev) [inline src/main/java/io/opentracing/contrib/cassandra/TracingSession.java]: please don't use stars in imports

File: src/main/java/io/opentracing/contrib/cassandra/TracingSession.java [hunk 2]
```diff
@@ -219,31 +220,56 @@ public ResultSetFuture executeAsync(Statement statement) {
...
    * {@inheritDoc}
    */
   @Override
   public PreparedStatement prepare(RegularStatement statement) {
-    return session.prepare(statement);
+    String query = getQuery(statement);
+    Span span = buildSpan(query);
+    try {
+      PreparedStatement resultSet = session.prepare(statement);
+      finishSpan(span);
```
Comment (malafeev) [inline src/main/java/io/opentracing/contrib/cassandra/TracingSession.java]: shouldn't span created (started) when prepared statement started to execute.
and span finished when prepared statement finished?
looks like here creation of statement is traced, but not execution.

Comment (tmszdmsk) [reply src/main/java/io/opentracing/contrib/cassandra/TracingSession.java]: ha! you're right ;)

I will :eyes: into it

**Comment by tmszdmsk** [comment]
After some research and debugging:
* `Mapper` creates prepared statements first and uses it in the next executions
* prepare statement span is created and has a proper parent
* consecutive operations on Mapper work and have proper parent span
* the problem is with the first operation which creates prepared statement and executes it straight away. execute operation is reported with no parent. I wasn't able to make it work.

**Comment by malafeev** [comment]
@tmszdmsk let's start from beginning.
Could you create a test which demonstrate problem you found? (Test should fail).
Then we will try to find solution.

**Comment by tmszdmsk** [comment]
@malafeev, sorry for the delay

I wrote a test that demonstrates the problem, it is actually why the tests are failing:
https://travis-ci.org/opentracing-contrib/java-cassandra-driver/builds/404851120#L3429

**Comment by malafeev** [comment]
@tmszdmsk I mean what is the initial problem? 
I don't think that we need to build span for prepared statement. (prepare* methods should not be changed).
Only when prepared statement is executed span should be created and finished. But it is already implemented in `public ResultSet execute(Statement statement)`.

**Comment by tmszdmsk** [comment]
@malafeev then this is the test case you want (in the first commit)
https://github.com/opentracing-contrib/java-cassandra-driver/pull/9/commits/aa6bf2caf0a4b60da5ab80b76e050bb3f9187372

**Comment by malafeev** [comment]
I'm closing this PR because it doesn't fix the issue.
There is no proper way to propagate active span into mapper's futures.
Most likely we need to create issue against Cassandra Mapper to add such kind support. 
Test is added to demonstrate issue.

**Pull request closed without merging**